pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTImportDeclaration.java

```diff
@@ -115,4 +128,20 @@ protected <P, R> R acceptVisitor(JavaVisitor<? super P, ? extends R> visitor, P
         return visitor.visit(this, data);
     }
 
+    void setModuleImport() {
+        moduleImport = true;
+    }
+
+    /**
+     * If this import declaration imports all the public top-level classes and interfaces
+     * of a module.
+     *
+     * @return {@code true} if this is a module declaration import
+     * @since 7.4.0
+     * @see <a href="https://openjdk.org/jeps/476">JEP 476: Module Import Declarations (Preview)</a> (Java 23)
+     */
+    @Experimental
+    public boolean isModuleImport() {
```

Comment:
Maybe name this just `isModule()` - since that it is a import declaration is already clear from the class name...

Reply:
I think it would read better with the old name, especially in xpath rules. The other getters in this class also already have Import in their name (eg `isImportOnDemand`/`@ImportOnDemand`) and it would be more consistent.

Reply:
I've renamed it back to `isModuleImport` now...